tools/volume_migration_scripts/README_migration_csi_to_cnsa.md

```diff
@@ -0,0 +1,158 @@
+# PV Migration Script – CSI → CNSA
+
+This script helps migrate existing Kubernetes **PersistentVolumes (PVs)** created under the **standalone CSI driver** to the **CNSA-compatible CSI driver** format.
+It ensures that workloads continue to access their data seamlessly after migration.
+
+## Key Features
+
+- Preserves all **original PV properties**, including:
+  - Reclaim policy (e.g., `Retain`, `Delete`)
+  - Access modes (`ReadWriteOnce`, `ReadWriteMany`, etc.)
+  - Storage capacity
+  - Filesystem type (`fsType`)
+  - Labels, annotations, and other PV metadata
+
+- Only the **volumeHandle path segment** is updated to CNSA’s required format.
+- Supports migration of PVs created from **different volume types** (fileset-based, CG, cache, static, dependent, independent).
+- Generates **backup YAML files** before applying changes.
+- Can be safely re-run if required (**idempotent migration**).
+
+## Why Migration is Required
+
+In the standalone **CSI driver** setup, PVs are created with a `volumeHandle` format tied to paths under **primary filesets**, different **volumeMounts**, or **consistency groups**.
+
+However, in a **CNSA setup**, this format becomes incompatible because:
+
+- The **primary fileset path** is no longer available.
```

Comment:
Do we need to mention that primary concept is no longer applicable in standalone CSI as well?